.github/workflows/precommit.yml

```diff
@@ -77,21 +77,53 @@ jobs:
           path: dist/
 
   test:
-    name: Test py-${{ matrix.os.download_name }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
+    name: Test py-${{ matrix.os }}-${{ matrix.python-version }}-${{ matrix.cloud-provider }}
     needs: build
-    runs-on: ${{ matrix.os.image_name }}
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - image_name: ubuntu-latest
-            download_name: linux
-          - image_name: macos-latest
-            download_name: macos
-          - image_name: windows-latest
-            download_name: windows
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        os: [macos-latest, windows-latest-64-cores, ubuntu-latest-64-cores]
+        python-version: ["3.9", "3.10", "3.11"]
```

Comment:
python 3.8 isn't eol until October. Should we keep it in the test matrix until then?

Reply:
3.8 is still tested in daily test job.

Reply:
We still have 3.8 test running every day https://github.com/snowflakedb/snowpark-python/blob/04b84a36e5405793fa96f7d2bb73a608424b2e0d/.github/workflows/daily_precommit.yml#L95